infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/protostream/OgmProtoStreamMarshaller.java

```diff
@@ -23,9 +23,7 @@
  */
 public final class OgmProtoStreamMarshaller extends ProtoStreamMarshaller {
 
-	//FIXME all static threadlocals should die
-	// See https://hibernate.atlassian.net/browse/OGM-1163
-	private static final ThreadLocal<SerializationContext> currentSerializationContext = new ThreadLocal<>();
+	private final ThreadLocal<SerializationContext> currentSerializationContext = new ThreadLocal<>();
```

Comment:
There's only one instance of `OgmProtoStreamMarshaller` for the whole Hot Rod Client.

That means all our parallel Session are using the same marshaller - yet concurrently I need it to instruct which schema to use for a specific IO operation.

A typical Infinispan client app would have a different Class registered in the marshaller for each entity; we don't have this luxury, or I'd have to bytecode-generate a Class definition for each "Table" we want to map.